lib/compress/zstd_cwksp.h

```diff
@@ -198,29 +210,120 @@ MEM_STATIC size_t ZSTD_cwksp_alloc_size(size_t size) {
 #endif
 }
 
+/**
+ * Returns an adjusted alloc size that is the nearest larger multiple of 64 bytes.
+ * Used to determine the number of bytes required for a given "aligned".
+ */
+MEM_STATIC size_t ZSTD_cwksp_aligned_alloc_size(size_t size) {
+    return ZSTD_cwksp_alloc_size(ZSTD_cwksp_align(size, ZSTD_CWKSP_ALIGNMENT_BYTES));
+}
+
+/**
+ * Returns the amount of additional space the cwksp must allocate
+ * for internal purposes (currently only alignment).
+ */
+MEM_STATIC size_t ZSTD_cwksp_slack_space_required(void) {
+    /* For alignment, the wksp will always allocate an additional n_1=[0, 63] bytes
+     * to align the beginning of tables section, as well as another n_2=[0, 63] bytes
+     * to align the beginning of the aligned secion.
+     * 
+     * Then, we also must allocate an additional tableExtraAlignmentBytes=(64 - n_1) +
+     * alignedExtraAlignmentBytes=(64 - n_2) bytes at the end of the workspace to ensure
+     * that we always use an additional 128 bytes in four separate allocations for alignment purposes.
+     */
+    size_t const slackSpace = 4*ZSTD_cwksp_alloc_size(ZSTD_CWKSP_ALIGNMENT_BYTES/2);
+    DEBUGLOG(5, "cwksp: slack space required: %zu", slackSpace);
+    return slackSpace;
+}
+
+/**
+ * Return the number of additional bytes required to align a pointer to the given number of bytes
+ */
+MEM_STATIC size_t ZSTD_cwksp_bytes_to_align_ptr(void* ptr, const size_t alignBytes) {
+    size_t const alignBytesMask = alignBytes - 1;
+    size_t const bytes = (alignBytes - ((size_t)ptr & (alignBytesMask))) & alignBytesMask;
+    return bytes;
```

Comment:
can you add an assert that `alignBytes` is a power of two (`assert((alignBits & alignBytesMask) == 0)`).